File: src/ros2_bridge/mappers/geometry_msgs/pose2_d.rs
//! Typed mapper for `geometry_msgs/msg/Pose2D`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn pose2_d_to_bus(
    msg: ros_env::geometry_msgs::msg::Pose2D,
) -> crate::geometry_msgs::msg::v1::Pose2D {
    crate::geometry_msgs::msg::v1::Pose2D {
        x: msg.x,
        y: msg.y,
        theta: msg.theta,
    }
}

pub(crate) fn pose2_d_to_ros(
    bus: crate::geometry_msgs::msg::v1::Pose2D,
) -> ros_env::geometry_msgs::msg::Pose2D {
    ros_env::geometry_msgs::msg::Pose2D {
        x: bus.x,
        y: bus.y,
        theta: bus.theta,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsPose2DMapper;

impl TypedTopicMapper for GeometryMsgsPose2DMapper {
    type Ros = ros_env::geometry_msgs::msg::Pose2D;
    type Bus = crate::geometry_msgs::msg::v1::Pose2D;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(pose2_d_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(pose2_d_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/pose2_d.py
"""Generated mapper for `geometry_msgs/msg/Pose2D`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.geometry_msgs.msg.v1 import Pose2D as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def pose2_d_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.x = msg.x
    bus.y = msg.y
    bus.theta = msg.theta
    return bus


def pose2_d_to_ros(bus):
    from geometry_msgs.msg import Pose2D as RosMsg

    out = RosMsg()
    out.x = bus.x
    out.y = bus.y
    out.theta = bus.theta
    return out


class GeometryMsgsPose2DMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from geometry_msgs.msg import Pose2D as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return pose2_d_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return pose2_d_to_ros(bus)
